### PR TITLE
Base Fix for hasMany issue  #4258

### LINKED ISSUE #4258: targetKey for hasMany and hasOne
I have two legacy tables in a 1:m association:
**Property**(id, statistical_id, name, ...), **VisitorStats**(id, statistical_id, total, ...)

The following works:

`VisitorStats.belongsTo(Property, { foreignKey: 'statistical_id', targetKey: 'statistical_id'} );`

Then, in the controller, I can query for VisitorStats and eager-load the Property for it.

However, I want to query the Property table and include the VisitorStats model. For this, I would have to do a hasMany association, like this:

`Property.hasMany(VisitorStats, { foreignKey: 'statistical_id' });`

**hasMany's doesn't support the targetKey parameter option, and automatically joins VisitorStats.statistical_id on Property.id when I query.** I expected the targetKey property to work like at a belongsTo association.

Is it possible to change the target key using hasMany? If not, how should I represent this 1:m association?

### PR DESCRIPTION
**Pull request by matthewhardern**
### Pull Request check-list

_Please make sure to review and check all of these items:_
- [ ] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [ ] Does your issue contain a link to existing issue (Closes #[issue]) or a description of the issue you are solving?
- [ ] Have you added new tests to prevent regressions?
- [ ] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [ ] Have you added an entry under `Future` in the changelog?

_NOTE: these things are not required to open a PR and can be done afterwards / while the PR is open._
### Description of change

This change is to enable defining a foreign key on the source table for hasMany function, that then in turn allows using a key that is not the primary key of that table. It also falls back on primary key when no sourceKey is provided.

Closes #4258

### CODE REVIEW

**Comment by janmeier** [comment]
Code looks good - But we need some tests as well :)

File: lib/associations/has-many.js [hunk 1]
```diff
@@ -272,6 +276,8 @@ HasMany.prototype.get = function(instances, options) {
     instances = undefined;
   }
 
+  console.log(association);
```
Comment (mickhansen) [inline lib/associations/has-many.js]: ^

**Comment by felixfbecker** [comment]
Thinking about this, we should then probably support both `sourceKey` and `targetKey` for `belongsToMany`. Which might get a bit confusing with `otherKey`. But there was an issue recently that would be solved by this.

**Comment by janmeier** [comment]
Fixed in https://github.com/sequelize/sequelize/pull/7113

**Pull request closed without merging**